libs/utils/compare-versions.js

```diff
@@ -0,0 +1,44 @@
+const indexOrEnd = (str, q) => str.indexOf(q) === -1 ? str.length : str.indexOf(q)
```

Comment:
Better use `compare-versions` directly other than copy it's code here

Reply:
The problem is that it throws an exception when a invalid semver version is detected and they are quit common on packagist. This was the only reason for not using it directly.

Reply:
Hmm, indeed a problem, but force comparing may cause unexpected exceptions too. What about filter versions with validation before compare?

Reply:
`semver` event provides it's own sorting function which would be ideal.
Maybe we need to sort out invalid versions first, sort them separately and put the correctly sorted (and semver valid) versions afterwards?

Reply:
@amio I will try to come up with a initial solution later today.

Reply:
Sure, no hurry ☕️